frost-core/src/tests.rs

```diff
@@ -59,6 +59,36 @@ pub fn check_sign_with_dealer<C: Ciphersuite, R: RngCore + CryptoRng>(mut rng: R
     check_sign(min_signers, key_packages, rng, pubkeys);
 }
 
+/// Check if corrupting a signature share makes aggregation fail.
+/// Part of [`check_sign`].
+fn check_corrupted_share<C: Ciphersuite + PartialEq, R: RngCore + CryptoRng>(
+    signature_shares: &[frost::round2::SignatureShare<C>],
+    mut rng: R,
+    signing_package: &frost::SigningPackage<C>,
+    pubkeys: &frost::keys::PublicKeyPackage<C>,
+) {
+    let identifier_one = 1u16.try_into().expect("should be nonzero");
+    let mut corrupted_signature_shares = signature_shares.to_owned();
+    let random_share = <<C::Group as Group>::Field>::random(&mut rng);
+    corrupted_signature_shares
+        .iter_mut()
+        .find(|share| share.identifier == identifier_one)
+        .unwrap()
+        .signature
+        .z_share = random_share;
+    let group_signature_res =
+        frost::aggregate(signing_package, &corrupted_signature_shares[..], pubkeys);
+    match group_signature_res {
+        Ok(_) => panic!("should fail"),
+        Err(Error::InvalidSignatureShare { signer }) => {
+            // starts_with is used instead of equality so that this works even with larger scalar fields
+            assert!(signer
+                .starts_with("0100000000000000000000000000000000000000000000000000000000000000"));
+        }
```

Comment:
Interesting; should we have a `Signer` type that impls a method that kinda wraps this up nicer?

Reply:
Ah yeah a new `Signer` type would overlap a lot with `Identifier`; what I was thinking was either way have a type that we could impl a comparison (either with bytes, or this sort of encoding as a string) explicitly (maybe impl `Eq` etc) vs having the signer be a `String` and calling `starts_with`. 

If we go with #183 we can just do a more fleshed out `impl Eq for Identifier` if we like, or a test-only / internal `impl Identifier<C> { pub fn starts_with(&self) -> bool { ... } }`, or something